src/store/transactions/selectors.ts

```diff
@@ -19,3 +20,15 @@ export const selectTransactionLastTxId = createSelector(
   selectTransactionsSlice,
   state => state.lastTxId
 );
+
+export const selectOriginContractSource = (contractName: string) =>
+  createSelector(selectTransactionsSlice, state => {
+    const address = contractName.split('.')[0];
+    const txs = Object.values(state.entities) as Transaction[];
```

Comment:
Would be nice to use `.values` but we can avoid the typecasting doing this, think it's worthwhile as its more type-safe.

```suggestion
    const txs = Object.keys(state.entities).map(tx_id => state.entities[tx_id]);
```